include/nuttx/compiler.h

```diff
@@ -154,6 +154,11 @@
 #    define farcall_function __attribute__((long_call))
 #  endif
 
+/* Branch prediction */
+
+#define likely(x) __builtin_expect((x), 1)
+#define unlikely(x) __builtin_expect((x), 0)
```

Comment:
need add dummy definition for other compiler

Reply:
done

Reply:
It will not work this way :D

Reply:
please follow how other macro is defined in this file, @Gary-Hobson